bug: v1.19.0 x86-32 binaries segmentation violate on first connection

## Bug Description

I got the proxy with

`wget https://dl.google.com/cloudsql/cloud_sql_proxy.linux.386 -O cloud_sql_proxy && chmod +x cloud_sql_proxy`

for months now and after your release yesterday I got the following error when I am using it inside my Python Scripts.

I run it inside gitlab:

```
Transform Prod SQL:
  image: python:3.7.2-alpine
  stage: Transform SQL
  variables:
    MYSQL_PWD: $SQL_PASS_PROD
  script:
    - wget https://dl.google.com/cloudsql/cloud_sql_proxy.linux.386 -O cloud_sql_proxy && chmod +x cloud_sql_proxy
    - ./cloud_sql_proxy -instances=xxxx=tcp:0.0.0.0:3306 -credential_file=client.json &
```

Error:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x804a0ac]
goroutine 102 [running]:
runtime/internal/atomic.Xadd64(0xa4780b4, 0x1, 0x0, 0x0, 0x0)
	/usr/local/go/src/runtime/internal/atomic/asm_386.s:105 +0xc
github.com/GoogleCloudPlatform/cloudsql-proxy/proxy/proxy.(*Client).handleConn(0xa478060, 0xffe56b9a, 0x34, 0x868f080, 0xa8ba090)
	/workspace/proxy/proxy/client.go:129 +0x50
created by github.com/GoogleCloudPlatform/cloudsql-proxy/proxy/proxy.(*Client).Run
	/workspace/proxy/proxy/client.go:120 +0x99
ERROR 2013 (HY000): Lost connection to MySQL server at 'handshake: reading initial communication packet', system error: 11

```

I'm able to replicate this on 64-bit linux with the 32-bit binary:
```sh
wget https://dl.google.com/cloudsql/cloud_sql_proxy.linux.386 -O cloud_sql_proxy && chmod +x cloud_sql_proxy
kvg@kvg:~$ ./start_proxy.sh $MYSQL_CONNECTION_NAME 
2020/11/20 08:50:01 current FDs rlimit set to 32768, wanted limit is 8500. Nothing to do here.
2020/11/20 08:50:03 Listening on 127.0.0.1:5432 for $MYSQL_CONNECTION_NAME 
2020/11/20 08:50:03 Ready for new connections
2020/11/20 08:52:35 New connection for "$MYSQL_CONNECTION_NAME "
panic: runtime error: invalid memory address or nil pointer dereference
```

But the 64-bit binary works just fine:
```
wget https://dl.google.com/cloudsql/cloud_sql_proxy.linux.386 -O cloud_sql_proxy && chmod +x cloud_sql_proxy
kvg@kvg:~$ ./start_proxy.sh $MYSQL_CONNECTION_NAME 
2020/11/20 08:59:04 current FDs rlimit set to 32768, wanted limit is 8500. Nothing to do here.
2020/11/20 08:59:05 Listening on 127.0.0.1:5432 for $MYSQL_CONNECTION_NAME 
2020/11/20 08:59:05 Ready for new connections
2020/11/20 08:59:07 New connection for "$MYSQL_CONNECTION_NAME "
2020/11/20 08:59:08 Instance $MYSQL_CONNECTION_NAME closed connection
```

Are you actually using an i386 arch? I'll have to see if I can find a compatible machine to test same architecture.

@kurtisvg @Carrotman42 Does this seem to be the same issue? Sorry I am not that well versed with this all. 

`root@raspberrypi:~/go/bin # ./cloud_sql_proxy -instances=*********:us-east4:algorithm-library=tcp:1234 -credential_file=./*********.json &
[1] 756
root@raspberrypi:~/go/bin # 2020/11/24 22:21:17 Rlimits for file descriptors set to {&{8500 1048576}}
2020/11/24 22:21:17 using credential file for authentication; email=devicetest2@sentinel-project-278421.iam.gserviceaccount.com
2020/11/24 22:21:43 Listening on 127.0.0.1:1234 for *********:us-east4:algorithm-library
2020/11/24 22:21:43 Ready fomysql -u sentinel_devices  -p --host 127.0.0.1 --port 1234
Enter password:
2020/11/24 22:22:01 New connection for "*********:us-east4:algorithm-library"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x12aec]

goroutine 42 [running]:
runtime/internal/atomic.goXadd64(0x247a654, 0x1, 0x0, 0x241c058, 0x0)
	/usr/local/go/src/runtime/internal/atomic/atomic_arm.go:103 +0x1c
github.com/GoogleCloudPlatform/cloudsql-proxy/proxy/proxy.(*Client).handleConn(0x247a600, 0xbed8f89b, 0x32, 0x669110, 0x24106e0)
	/home/pi/go/pkg/mod/github.com/!google!cloud!platform/cloudsql-proxy@v1.19.0/proxy/proxy/client.go:129 +0x4c
created by github.com/GoogleCloudPlatform/cloudsql-proxy/proxy/proxy.(*Client).Run
	/home/pi/go/pkg/mod/github.com/!google!cloud!platform/cloudsql-proxy@v1.19.0/proxy/proxy/client.go:120 +0x88
ERROR 2013 (HY000): Lost connection to MySQL server at 'reading initial communication packet', system error: 0 "Internal error/check (Not system error)"`

--- a/proxy/proxy/client.go
+++ b/proxy/proxy/client.go
@@ -60,6 +60,13 @@ type CertSource interface {
 // Client is a type to handle connecting to a Server. All fields are required
 // unless otherwise specified.
 type Client struct {
+	// ConnectionsCounter is used to enforce the optional maxConnections limit
+	ConnectionsCounter uint64
+
+	// MaxConnections is the maximum number of connections to establish
+	// before refusing new connections. 0 means no limit.
+	MaxConnections uint64
+
 	// Port designates which remote port should be used when connecting to
 	// instances. This value is defined by the server-side code, but for now it
 	// should always be 3307.
@@ -95,13 +102,6 @@ type Client struct {
 	// RefreshCertBuffer is the amount of time before the configuration expires to
 	// attempt to refresh it. If not set, it defaults to 5 minutes.
 	RefreshCfgBuffer time.Duration
-
-	// MaxConnections is the maximum number of connections to establish
-	// before refusing new connections. 0 means no limit.
-	MaxConnections uint64
-
-	// ConnectionsCounter is used to enforce the optional maxConnections limit
-	ConnectionsCounter uint64
 }
 
 type cacheEntry struct {

--- a/proxy/proxy/client_test.go
+++ b/proxy/proxy/client_test.go
@@ -25,6 +25,7 @@ import (
 	"sync/atomic"
 	"testing"
 	"time"
+	"unsafe"
 )
 
 const instance = "instance-name"
@@ -283,5 +284,13 @@ func TestRefreshTimer(t *testing.T) {
 	}
 	if !newCfg.lastRefreshed.After(cfg.lastRefreshed) {
 		t.Error("expected cert to be refreshed.")
+	}
+}
+
+func TestSyncAtomicAlignment(t *testing.T) {
+	// The sync/atomic pkg has a bug that requires the developer to guarantee 64-bit alignment when using 64-bit functions on 32-bit systems. 
+	c := &Client{}
+	if a := unsafe.Offsetof(c.ConnectionsCounter); a%64 != 0 {
+		t.Errorf("Client.ConnectionsCounter is not aligned: want %v, got %v", 0, a)
 	}
 }

--- a/proxy/proxy/client_test.go
+++ b/proxy/proxy/client_test.go
@@ -288,7 +288,7 @@ func TestRefreshTimer(t *testing.T) {
 }
 
 func TestSyncAtomicAlignment(t *testing.T) {
-	// The sync/atomic pkg has a bug that requires the developer to guarantee 64-bit alignment when using 64-bit functions on 32-bit systems. 
+	// The sync/atomic pkg has a bug that requires the developer to guarantee 64-bit alignment when using 64-bit functions on 32-bit systems.
 	c := &Client{}
 	if a := unsafe.Offsetof(c.ConnectionsCounter); a%64 != 0 {
 		t.Errorf("Client.ConnectionsCounter is not aligned: want %v, got %v", 0, a)

